sdk/core/Azure.Core/src/Pipeline/Internal/TelemetryPolicy.cs

```diff
@@ -7,24 +7,23 @@ namespace Azure.Core.Pipeline
 {
     internal class TelemetryPolicy : HttpPipelineSynchronousPolicy
     {
-        private readonly string _header;
+        private readonly string _defaultHeader;
 
-        public TelemetryPolicy(string componentName, string componentVersion, string? applicationId)
+        public TelemetryPolicy(TelemetryPackageInfo packageInfo)
         {
-            var platformInformation = $"({RuntimeInformation.FrameworkDescription}; {RuntimeInformation.OSDescription})";
-            if (applicationId != null)
+            _defaultHeader = packageInfo.UserAgentValue;
+        }
+
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            if (message.TryGetInternalProperty(typeof(TelemetryPackageInfo), out var userAgent))
```

Comment:
Since we want to make this property accessible just by telemetry policy, we should not use a public type as the key in the dictionary (once we make the setting APIs public)

Reply:
We make the type accessible publicly either way by exposing a public method to allow the values to be set. Did you have an approach in mind that would both allow public setting of the property while preventing access to it?